### PR TITLE
Add API support for queries for protocol params.

### PR DESCRIPTION
**Pull request by dcoutts**
Several kinds of params:
1. current values of updateable protocol params
2. any pending proposed param updates
3. the non-updateable params fixed in the (Shelley) genesis file

### CODE REVIEW

**Comment by dcoutts** [comment]
The first two patches are from PR 2238. This can be rebased once they're merged.

**Review by newhoggy — APPROVED** [review]

File: cardano-api/src/Cardano/Api/Query.hs [hunk 1]
```diff
@@ -105,18 +115,17 @@ data QueryInShelleyBasedEra era result where
...
 
      QueryUTxO
        :: Maybe (Set AddressAny)
        -> QueryInShelleyBasedEra era (UTxO era)
 
---     QueryStakeAddresses
---       :: Set StakeAddress
---       -> QueryInShelleyBasedEra (Map StakeAddress Lovelace,
---                                  Map StakeAddress PoolId)
+     QueryStakeAddresses
```
Comment (Jimbo4350) [inline cardano-api/src/Cardano/Api/Query.hs:125]: @dcoutts  We are duplicating work again: https://github.com/input-output-hk/cardano-node/commit/535504320840194f1a0701ed39c2bc6e6bd575b2

**Review by Jimbo4350 — APPROVED** [review]

File: cardano-api/src/Cardano/Api/Query.hs (hunk 1 above)
Comment (dcoutts) [reply cardano-api/src/Cardano/Api/Query.hs:125]: Oops. You can take over this PR.

**Comment by Jimbo4350** [comment]
bors merge

**Pull request merged**